rust/datafusion/src/execution/context.rs

```diff
@@ -160,6 +161,21 @@ impl ExecutionContext {
 
                 Ok(Rc::new(RefCell::new(rel)))
             }
+            LogicalPlan::Limit {
+                ref expr,
+                ref input,
+                ..
+            } => {
+                let input_rel = self.execute(input)?;
+
+                let input_schema = input_rel.as_ref().borrow().schema().clone();
+
+                let compiled_expr = compile_scalar_expr(&self, expr, &input_schema)?;
```

Comment:
The LIMIT expression will be a literal e.g. `LIMIT 10` so I don't think we need to compile and evaluate the expression but just use pattern matching to extract the number.

```
match expr {
        &Expr::Literal(ref value) => /* extract value */,
        _ => /* error - only literals are supported */
}
```
``